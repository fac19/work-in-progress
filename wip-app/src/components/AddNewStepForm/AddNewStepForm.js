import React from "react";
import { Button } from "@material-ui/core";
import { NewStepSection, NewStepForm } from "./AddNewStepForm.style";
// import Upload from "../Upload/Upload";
import { postStep } from "../../utils/post-fetch";
import { postImage } from "../../utils/ext-fetch";
import getB64 from "../../utils/b64";

const AddNewStepForm = ({ projectId }) => {
  const [form, setForm] = React.useState({
    step_name: "",
    step_description: "",
    step_link: "",
  });
  const [file, uploadFile] = React.useState();
  const handleChange = (event) => {
    const { name, value } = event.target;
    setForm({ ...form, [name]: value });
  };

  const handleUpload = (event) => {
    uploadFile(event.target.files[0]);
    console.log(
      "AddNewStepForm -> event.target.files[0]",
      event.target.files[0]
    );
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    getB64(file)
      .then(postImage)
      .then((data) => setForm({ ...form, [step_link]: data.value }));
    console.log(form);
    // upload media to cloudinary
    // then save new step_link into form
    // post request add new step
    // history.push("/project/projectId"))
  };

  return (
    <NewStepSection>
      <h2>Add Step to Project</h2>
      <NewStepForm onSubmit={handleSubmit}>
        <label htmlFor="step_name">Step Name</label>
        <input
          type="text"
          id="step_name"
          name="step_name"
          onChange={handleChange}
          required
        />
        <label htmlFor="step_description">Step Description</label>
        <input
          type="text"
          id="step_description"
          name="step_description"
          onChange={handleChange}
          required
        />
        <label htmlFor="step_description">Step Link:</label>
        <input
          type="file"
          id="step_link"
          name="step_link"
          onChange={handleUpload}
          required
        />

        {/* <Upload setStepLink={setStepLink} /> */}
        <Button variant="contained" color="secondary" type="submit">
          Save Step To Project
        </Button>
      </NewStepForm>
    </NewStepSection>
  );
};

export default AddNewStepForm;
