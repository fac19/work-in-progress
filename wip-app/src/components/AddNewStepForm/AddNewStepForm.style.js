import styled from "styled-components";

const NewStepSection = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const NewStepForm = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export { NewStepSection, NewStepForm };
