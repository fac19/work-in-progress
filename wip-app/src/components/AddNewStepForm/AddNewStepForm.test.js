import React from "react";
import TestRenderer from "react-test-renderer";
import AddNewStepForm from "./AddNewStepForm";

describe("Snapshot test", () => {
  it("renders the correct ui", () => {
    const component = TestRenderer.create(<AddNewStepForm />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
