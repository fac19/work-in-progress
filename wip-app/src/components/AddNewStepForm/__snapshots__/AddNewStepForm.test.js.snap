// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<section
  className="sc-AxjAm lcVokx"
>
  <h2>
    Add Step to Project
  </h2>
  <form
    className="sc-AxirZ dcYQlC"
    onSubmit={[Function]}
  >
    <label
      htmlFor="step_name"
    >
      Step Name
    </label>
    <input
      id="step_name"
      name="step_name"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      htmlFor="step_description"
    >
      Step Description
    </label>
    <input
      id="step_description"
      name="step_description"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      htmlFor="step_description"
    >
      Step Link:
    </label>
    <input
      id="step_link"
      name="step_link"
      onChange={[Function]}
      required={true}
      type="file"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Save Step To Project
      </span>
    </button>
    <p>
      
    </p>
  </form>
</section>
`;
